stats/dogstatsd/config.go

```diff
@@ -0,0 +1,87 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2016 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package dogstatsd
+
+import (
+	"encoding/json"
+
+	statsd "github.com/loadimpact/k6/stats/statsd/shared"
+)
+
+// ConfigFields contains statsd configuration
+type ConfigFields struct {
+	Addr         string `json:"addr,omitempty" envconfig:"DOGSTATSD_ADDR"`
+	P            string `json:"port,omitempty" envconfig:"DOGSTATSD_PORT" default:"8126"`
+	BuffSize     int    `json:"buffer_size,omitempty" envconfig:"DOGSTATSD_BUFFER_SIZE" default:"20"`
+	Namespace    string `json:"namespace,omitempty" envconfig:"DOGSTATSD_NAMESPACE"`
+	TagWhitelist string `json:"tag_whitelist,omitempty" envconfig:"DOGSTATSD_TAG_WHITELIST" default:"status, method"`
+}
+
+// Config defines a config type
+type Config ConfigFields
+
+// Address returns client address
+func (c Config) Address() string {
```

Comment:
Hm, why are these accessors? They don't seem to do anything.

Reply:
They are being used to normalise StatsD and DogStatsD config. 
https://github.com/uswitch/k6/blob/87925ec060a89da573f90b461f6198b378a6b357/stats/statsd/shared/client.go#L33

Reply:
Couldn't you use the same struct and just ignore the fields one doesn't use?

Reply:
I could but they refer to different ENV's. https://github.com/uswitch/k6/blob/cf72b91566cbd1dc886ad58ef0ff8f60c215f420/stats/statsd/config.go#L31 vs https://github.com/uswitch/k6/blob/cf72b91566cbd1dc886ad58ef0ff8f60c215f420/stats/dogstatsd/config.go#L31

Reply:
Possible workaround:

```go
// package shared
type Config {
  Addr string `envconfig:"ADDR"`
}

// package statsd
type Config struct {
  shared.Config `envconfig:"STATSD"`
}

// package dogstatsd
type Config struct {
  shared.Config `envconfig:"DOGSTATSD"`
}
```

Reply:
hmm not sure about this. Just feels a bit weird and the interface solutions looks cleaner. What do you think the trade-off is here?

Reply:
Your version is… really verbose, and it's essentially solving a parsing problem by adapting the data, which feels backwards to me.

Reply:
You could also do something like just setting `envconfig:"ADDR"` and then using a different prefix in `envconfig.Process()`.

Reply:
I agree, is really verbose. I think using process is a good idea, let me give it a go.